### PR TITLE
feat: Filter Storage Mode Responses by Submission Id

### PR DESCRIPTION
**Pull request by frankchn**
## Problem

Users have trouble finding specific responses and trying to download attachments for these responses.

## Solution

This introduces a new textbox which filters submissions by reference ID so users can jump to a certain response immediately.

## Screenshots

image

image

image

### CODE REVIEW

**Comment by jonathangohjy** [comment]
Have uploaded screens on Zeplin for your review

[Placement of search bar](https://app.zeplin.io/project/5b29ffde48eaaf6318d01b05/screen/5f2b713ca4a638880b1ddcdb)
No Results [screen](https://app.zeplin.io/project/5b29ffde48eaaf6318d01b05/screen/5f2b713d06474f291cc888f)

**Comment by frankchn** [comment]
@liangyuanruo can you take a look? this is pretty close -- I just don't have the SVG for the "filter" icon.

**Comment by liangyuanruo** [comment]
Hey @frankchn ,

I just tried it out the feature, and noticed a few things

1. I see @jonathangohjy has introduced the filter button, but I would like to reverting back to the implementation without the filter button instead. The reason is because we expect users to copy-paste the ObjectID into the input box from the downloaded CSV, and we want the search to be immediately responsive, instead of introducing another click for the user interaction - I'd suggest using the `ng-change` directive here with the `<input>` element. We'd also need to introduce a debounce (I believe `ng-model-options` provide that out of the box).

2. File upload functionality appears to have been broken, although this may be to do with another PR related to CSP headers - could you rebase and see if it fixes the issue?

**Comment by frankchn** [comment]
@liangyuanruo Thanks for the comments! I updated the functionality per your suggestion in (1) and tested (2) to verify that file upload works with the latest rebase.

File: src/app/controllers/encrypt-submissions.server.controller.js [hunk 1]
```diff
@@ -202,15 +202,25 @@ exports.saveResponseToDb = function (req, res, next) {
  */
 exports.getMetadata = function (req, res) {
   let pageSize = 10
-  let { page } = req.query || {}
+  let { page, filterBySubmissionRefId } = req.query || {}
```
Comment (liangyuanruo) [inline src/app/controllers/encrypt-submissions.server.controller.js:205]: if i could make a small nitpick here, which is to rename this query parameter to `submissionId` instead?

File: src/public/modules/forms/services/submissions.client.factory.js [hunk 2]
```diff
@@ -143,10 +143,14 @@ function SubmissionsFactory(
     },
     getMetadata: function (params) {
       const deferred = $q.defer()
-      const resUrl = `${fixParamsToUrl(params, submitAdminUrl)}/metadata?page=${
+      let resUrl = `${fixParamsToUrl(params, submitAdminUrl)}/metadata?page=${
         params.page
       }`
 
+      if (params.filterBySubmissionRefId) {
+        resUrl += `&filterBySubmissionRefId=${params.filterBySubmissionRefId}`
```
Comment (liangyuanruo) [inline src/public/modules/forms/services/submissions.client.factory.js:151]:
```suggestion
        resUrl += `&submissionId=${params.filterBySubmissionRefId}`
```

File: src/app/controllers/encrypt-submissions.server.controller.js (hunk 1 above)
Comment (frankchn) [reply src/app/controllers/encrypt-submissions.server.controller.js:205]: Cool, will rename.

File: src/public/modules/forms/services/submissions.client.factory.js (hunk 2 above)
Comment (frankchn) [reply src/public/modules/forms/services/submissions.client.factory.js:151]: Will rename

**Comment by frankchn** [comment]
OK I guess if MongoDB isn't quite smart enough to use the index, we will just have to do a single item query.

**Comment by frankchn** [comment]
@liangyuanruo Tested on staging and both attachment upload and filtering should work now that we switched to  `Submission.findOne` instead of using the same mechanism.

**Pull request merged**